core/trino-main/src/main/java/io/trino/execution/scheduler/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -1190,6 +1195,23 @@ private void createStageExecution(SubPlan subPlan, boolean rootFragment, Map<Sta
             }
         }
 
+        private boolean isNoMemoryFragment(PlanFragment fragment)
+        {
+            // If source fragments are not tagged as "no-memory" assume that they may produce significant amount of data.
+            // We stay on the safe side an assume that we should use standard memory estimation for this fragment
+            if (!fragment.getRemoteSourceNodes().stream().flatMap(node -> node.getSourceFragmentIds().stream())
+                    .allMatch(sourceFragmentId -> stageExecutions.get(getStageId(sourceFragmentId)).isNoMemoryFragment())) {
+                return false;
+            }
+
+            // If fragment source is not reading any external tables or only accesses information_schema assume it does not need significant amount of memory.
+            // Allow scheduling even if whole server memory is pre allocated.
+            List<PlanNode> tableScanNodes = PlanNodeSearcher.searchFrom(fragment.getRoot()).whereIsInstanceOfAny(TableScanNode.class).findAll();
+            return tableScanNodes.stream()
+                    .map(node -> (TableScanNode) node)
+                    .allMatch(node -> node.getTable().getConnectorHandle() instanceof InformationSchemaTableHandle);
```

Comment:
If we want this to work for JDBC, we could do
```java
.allMatch(handle -> (node.getTable().getConnectorHandle() instanceof InformationSchemaTableHandle)) || 
        ((node.getTable().getCatalogHandle().getCatalogName().equals(GlobalSystemConnector.NAME) &&
         (node.getTable().getConnectorHandle() instanceof SystemHandle systemHandle) &&
        systemHandle.getSchemaName().equals("jdbc")))

Reply:
👍